minari/dataset/_storages/arrow_storage.py

```diff
@@ -257,8 +264,11 @@ def _decode_info(values: pa.Array):
     return nested_dict
 
 
-class NumpyEncoder(json.JSONEncoder):
-    def default(self, obj):
-        if isinstance(obj, np.ndarray):
-            return obj.tolist()
-        return super().default(obj)
+def encode_info_list(info_list: List[Dict[str, Any]]) -> pa.Array:
+    serialized_list = [serialize_dict(d) for d in info_list]
+    return pa.array(serialized_list, type=pa.string())
```

Comment:
serializing as dict and using JSON to serialize/deserialize is rather inefficient

Reply:
The list(dict) option is now an optional feature of DataCollector with optional arg infos_format wich defaults to "dict" when None or "dict and use list(dict) when infos_format="list". For the rest the all the functionality is the same just for this optional behavior.

Reply:
Also, if the user has the orjson library installed or installs it Minari will use this library for fast serde when available in case optinal feature of infos_format="list"